include/utils/faultTree.h

```diff
@@ -0,0 +1,52 @@
+#ifndef _FAULT_TREE_H
+#define _FAULT_TREE_H
+
+#include "all.h"
+#include "parser.h"
```

Comment:
You can also replace parser with existing CSV reader utilities in MOOSE. You can find a sample usage in the Fragility Vectorpostprocessor which reads . This is at src/vectorpostprocessors/Fragility.C. Please refer to the calcDemandsFromFile method in this file - specifically, lines 232 - 235

Reply:
Hi Chandu,

We have one problem here. The existing parser in Mastodon converts csv information into double type. But in our fault tree code, we have logic.csv file which has information about fault tree logic (event names, operators (AND/OR)). Can you please point us to any parser which outputs string type?

Reply:
HI @ssbodda, you're right, DelimitedReader is only meant for type Real data with text headers. In that case, I think it is better to do the parsing by yourself. Instead of having a separate parser class, I recommend you parse it in one of the methods in the faulttreeanalysis file. The filename still has to go in as an input, but we can deal with that later. If possible, I even recommend you use some predefined values for the faulttree logic and the basic events and we can tell you about parsing later. For now, if you can get the unit tests working, that would be great.